### PR TITLE
Add boring crypto build to release

### LINKED ISSUE #4133: Add boringcrypto support for linux x64/arm64 binary builds
### Request

For usage in more locked down environments the Agent should build binaries for linux amd64/arm64 utilizing boring crypto

### Use case

for usage in environments where boring crypto is necessary

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

Add FIPS (boringcrypto) targets for linux amd64/arm64 along with docker images. This closes #4133

#### Notes to the Reviewer

Manually checked the builds via `go tool nm ./grafana-agent-linux-amd64-boringcrypto| grep FIPS` though you have to not strip symbols to see these.  The output of checking for boringcrypto is the below.

```
4dfbb80 t BORINGSSL_FIPS_abort
 4df7a80 t FIPS_mode_set
 4df7a90 t FIPS_read_counter
 4d9a8d0 T _cgo_d8b1bdd8e714_Cfunc__goboringcrypto_FIPS_mode
 4df7a70 t _goboringcrypto_FIPS_mode
  57c8c0 t crypto/internal/boring._Cfunc__goboringcrypto_FIPS_mode.abi0
 ad18640 d crypto/internal/boring._cgo_d8b1bdd8e714_Cfunc__goboringcrypto_FIPS_mode
  579680 t crypto/internal/boring/sig.FIPSOnly.abi0
 ad1ce80 d crypto/tls.defaultCipherSuitesFIPS
 ad1cf00 d crypto/tls.defaultFIPSCurvePreferences
```

Checking against the non boring crypto does not show any of the above and only shows a few references in the AWS SDK regarding FIPs endpoints.

Tested the images against my mrdgrafana account to ensure they worked as expected. 

#### PR Checklist

- [X] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by clayton-cornell — COMMENTED** [review]
Docs look OK

File: .drone/drone.yml [hunk 1]
```diff
@@ -878,6 +878,40 @@ trigger:
 type: docker
 ---
 kind: pipeline
+name: Build agent-boringcrypto (Linux amd64 boringcrypto)
```
Comment (erikbaranowski) [inline .drone/drone.yml:881]: nit: consider moving under `Build agent-flow (FreeBSD amd64)` so all the agent builds are grouped together

**Review by erikbaranowski — APPROVED** [review]
LGTM, 1 small nit for consideration

File: .drone/drone.yml (hunk 1 above)
Comment (erikbaranowski) [reply .drone/drone.yml:881]: ah right, we have the jsonnet generating this file. no worries it was a small nit. not worth spending much time on.

**Review by rfratto — APPROVED** [review]
LGTM, gave it a thorough look but nothing stood out to me

**Pull request merged**